test/chainrules.jl

```diff
@@ -1,17 +1,19 @@
 @testset "chainrules.jl" begin
-    @testset "BlockDiagonal" begin
-        x = [randn(1, 2), randn(2, 2)]
-        test_rrule(BlockDiagonal, x)
-    end
+    @testset for V in (Tuple, Vector)
+        @testset "BlockDiagonal" begin
+            x = V([randn(1, 2), randn(2, 2)])
+            test_rrule(BlockDiagonal, x)
+        end
 
-    @testset "Matrix" begin
-        D = BlockDiagonal([randn(1, 2), randn(2, 2)])
-        test_rrule(Matrix, D)
-    end
+        @testset "Matrix" begin
+            B = BlockDiagonal(V([randn(1, 2), randn(2, 2)]))
+            test_rrule(Matrix, B)
```

Comment:
When the blocks are a tuple, this errors with
```julia
Matrix: Error During Test at /Users/npr/repos/BlockDiagonals.jl/test/chainrules.jl:8
  Got exception outside of a @test
  MethodError: no method matching +(::Composite{Tuple{Array{Float64,2},Array{Float64,2}},Tuple{Array{Float64,2},Array{Float64,2}}}, ::Array{Array{Float64,2},1})
```

Reply:
changed so this PR just doesn't define any `rrule` methods for the `blocks::Tuple` case (see https://github.com/invenia/BlockDiagonals.jl/pull/56#discussion_r576447746)